components/brave_wallet_ui/components/buy-send-swap/buy/index.tsx

```diff
@@ -67,15 +67,16 @@ export const Buy = ({
 
   // memos
   const supportedBuyOptions: BuyOption[] = React.useMemo(() => {
+    const onRampAssetMap = {
+      [BraveWallet.OnRampProvider.kWyre]: wyreAssetOptions,
+      [BraveWallet.OnRampProvider.kRamp]: rampAssetOptions,
+      [BraveWallet.OnRampProvider.kSardine]: sardineAssetOptions
+    }
```

Comment:
Non blocking: 
Are we ok with the performance trade-off of using an object lookup vs leaving this as a `switch`?

example benchmark: https://jsben.ch/JYZLQ

Reply:
@josheleonard This change was motivated by improving readability of the code especially when more cases need to be checked in the future